pig/src/main/java/org/apache/iceberg/pig/IcebergPigInputFormat.java

```diff
@@ -246,7 +246,11 @@ private boolean advance() throws IOException {
     private Object convertPartitionValue(Type type, Object value) {
       if (type.typeId() == Types.BinaryType.get().typeId()) {
         ByteBuffer buffer = (ByteBuffer) value;
-        return new DataByteArray(buffer.get(new byte[buffer.remaining()]).array());
+        byte[] bytes = new byte[buffer.remaining()];
+        buffer.get(bytes);
+        // Return the input buffer back to its original position.
+        buffer.position(buffer.position() - bytes.length);
+        return new DataByteArray(bytes);
```

Comment:
I updated this one as you requested @rdblue, however I wasn't sure if I should return the input `value` to its original position (vs consuming the remainder of the byte buffer and moving the position to the end).

Reply:
Looks good, but we should use `slice` or `duplicate` to read to avoid needing to restore the original buffer's state.

Reply:
Ok. I will update this one to use `slice` to avoid having to restore the original buffer's state.